semgrep-core/src/engine/Dataflow_tainting.ml

```diff
@@ -334,12 +341,20 @@ let check_function_signature env fun_exp args_taint =
       Some
         (fun_sig
         |> List.filter_map (function
-             | SrcToReturn dm ->
-                 let dm = Call (eorig, dm) in
-                 Some (Taint.singleton (Src dm))
-             | ArgToReturn i -> taint_of_arg i
-             | ArgToSink (i, sink) ->
-                 let sink = Call (eorig, sink) in
+             | SrcToReturn (dm, trace) ->
+                 let dm = Call (eorig, trace, dm) in
```

Comment:
Nit: why is this "trace, dm" but SrcToReturn "dm, trace"? If we don't make it a record, would be better to have an order on the args

Reply:
There is a meaning to it, but I understand it's not clear. In `SrcToReturn` it is the trace _from_ the source `dm` to the `return` (it's missing something to identify the `return`... but I'll add a `tok` to fix that!). In `Call` is the trace from the function in `eorig` _to_  the match `dm`. So I tried to put the `trace` in between the "start" and "end" of the "path".